src/app/organization/offices/edit-office/edit-office.component.ts

```diff
@@ -0,0 +1,81 @@
+/** Angular Imports */
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { DatePipe } from '@angular/common';
+
+/** Custom Services */
+import { OrganizationService } from 'app/organization/organization.service';
+
+/**
+ * Edit Office component.
+ */
+@Component({
+  selector: 'mifosx-edit-office',
+  templateUrl: './edit-office.component.html',
+  styleUrls: ['./edit-office.component.scss']
+})
+export class EditOfficeComponent implements OnInit {
+
+  /** Selected Data. */
+  officeData: any;
+  /** Office form. */
+  officeForm: FormGroup;
+  /** Minimum Date allowed. */
+  minDate = new Date(2000, 0, 1);
+  /** Maximum Date allowed. */
+  maxDate = new Date();
+
+    /**
+     * Retrieves the charge data from `resolve`.
+     * @param {ProductsService} organizationService Organization Service.
+     * @param {FormBuilder} formBuilder Form Builder.
+     * @param {ActivatedRoute} route Activated Route.
+     * @param {Router} router Router for navigation.
+     * @param {MatDialog} dialog Dialog reference.
+     * @param {DatePipe} datepipe Convert Date.
+     */
+    constructor(private organizationService: OrganizationService,
+                private formBuilder: FormBuilder,
+                private route: ActivatedRoute,
+                private router: Router,
+                private datepipe: DatePipe) {
+      this.route.data.subscribe((data: { officeTemplate: any }) => {
+        this.officeData = data.officeTemplate;
+      });
+    }
+
+  ngOnInit() {
+    this.createOfficeForm();
+  }
+
+  /**
+   * Create Edit Office Form.
+   */
+  createOfficeForm() {
+    this.officeForm = this.formBuilder.group({
+      'name': [this.officeData.name, Validators.required],
+      'parentId': [this.officeData.parentId, Validators.required],
+      'openingDate': [this.officeData.openingDate && new Date(this.officeData.openingDate), Validators.required],
+      'externalId': [this.officeData.externalId],
+    });
```

Comment:
add if statement:
if officeData.allowed parents {
officeform.addcontrol ('parentId': [this.officeData.parentId, Validators.required])
}

Reply:
Can allowed parents be empty? Is this if statement really necessary?

Reply:
the allowed parents array should not be empty add necessary if statement for that

Reply:
Changes added. Please have a look